Il2CppInterop.Generator/Contexts/RewriteGlobalContext.cs

```diff
@@ -28,18 +25,22 @@ public class RewriteGlobalContext : IDisposable
         GameAssemblies = gameAssemblies;
         UnityAssemblies = unityAssemblies;
 
+        Il2CppAssemblyResolver assemblyResolver = new();
+
         foreach (var sourceAssembly in gameAssemblies.Assemblies)
         {
-            var assemblyName = sourceAssembly.Name.Name;
+            var assemblyName = sourceAssembly.Name!;
             if (assemblyName == "Il2CppDummyDll")
             {
-                sourceAssembly.Dispose();
                 continue;
             }
 
-            var newAssembly = AssemblyDefinition.CreateAssembly(
-                new AssemblyNameDefinition(sourceAssembly.Name.Name.UnSystemify(options), sourceAssembly.Name.Version),
-                sourceAssembly.MainModule.Name.UnSystemify(options), sourceAssembly.MainModule.Kind);
+            var newAssembly = new AssemblyDefinition(sourceAssembly.Name.UnSystemify(options), sourceAssembly.Version);
+            var newModule = new ModuleDefinition(sourceAssembly.ManifestModule?.Name.UnSystemify(options), KnownCorLibs.MsCorLib_v2_0_0_0);
```

Comment:
I don't know. I chose mscorlib `2.0` because it makes reference resolving work automatically in the generation process, whereas I think I remember encountering issues with `4.0`.

Reply:
The target framework of the interop assemblies should probably be the same as `Il2CppInterop.Runtime` (so currently net6.0, but we really should upgrade that), that way we can use all the modern apis in generated code.

Reply:
I changed it over to `System.Runtime`.